internal/v2/models/bundle.go

```diff
@@ -32,3 +36,95 @@ type BundleV2Model struct {
 	Name          types.String              `tfsdk:"name"`
 	AccessTargets []BundleAccessTargetModel `tfsdk:"access_targets"`
 }
+
+func BundleResponseToModel(ctx context.Context, response client.BundlePublicV2Model) (*BundleV2Model, error) {
+	model := BundleV2Model{
+		ID:   types.StringValue(response.ID),
+		Name: types.StringValue(response.Name),
+	}
+
+	accessTargets, err := convertBundleAccessTargetsToModel(ctx, response.AccessTargets)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert access targets: %w", err)
+	}
+	model.AccessTargets = accessTargets
+
+	return &model, nil
+}
+
+func BundleModelToUpsertRequest(ctx context.Context, model BundleV2Model) (*client.UpsertBundlePublicV2Model, error) {
+	upsert := client.UpsertBundlePublicV2Model{
+		Name: model.Name.ValueString(),
+	}
+
+	targets, err := convertBundleAccessTargetsToUpsertRequest(ctx, model.AccessTargets)
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert access targets: %w", err)
+	}
+
+	upsert.AccessTargets = targets
+
+	return &upsert, nil
+}
+
+func convertBundleAccessTargetsToModel(ctx context.Context, accessTargets []client.AccessBundleAccessTargetPublicV2Model) ([]BundleAccessTargetModel, error) {
+	var modelTargets []BundleAccessTargetModel
+
+	for _, target := range accessTargets {
+		modelTarget := BundleAccessTargetModel{}
+
+		if val, ok := target.Integration.Get(); ok {
+			integrationTarget, err := convertIntegrationTargetToModel(ctx, val)
+			if err != nil {
+				return nil, fmt.Errorf("failed to convert integration target: %w", err)
+			}
+			modelTarget.Integration = integrationTarget
+		}
+
+		if val, ok := target.AccessScope.Get(); ok {
+			modelTarget.AccessScope = &AccessScopeTargetModel{
+				Name: types.StringValue(val.AccessScopeName),
+			}
+		}
+
+		modelTargets = append(modelTargets, modelTarget)
+	}
+
+	return modelTargets, nil
+}
+
+func convertBundleAccessTargetsToUpsertRequest(ctx context.Context, models []BundleAccessTargetModel) ([]client.AccessBundleAccessTargetUpsertPublicV2Model, error) {
+	var targets []client.AccessBundleAccessTargetUpsertPublicV2Model
+
+	for i, model := range models {
+		target := client.AccessBundleAccessTargetUpsertPublicV2Model{}
+		setCount := 0
```

Comment:
It is easier to enforce either access scope or integration, but not both or none, via the code here.
Doing it via the Terraform schema verification is more challenging due to the complexity of the schema,